yt_dlp/extractor/viu.py

```diff
@@ -394,3 +397,146 @@ def download_playback():
             'formats': formats,
             'subtitles': subtitles,
         }
+
+
+class ViuOTTNewBaseIE(InfoExtractor):
+    _BASE_QUERY = {
+        'ver': 1.0,
+        'fmt': 'json',
+        'aver': 5.0,
+        'appver': 2.0,
+        'appid': 'viu_desktop',
+        'platform': 'desktop',
+    }
+
+    _DEVICE_ID = str(uuid.uuid4())
+    _SESSION_ID = str(uuid.uuid4())
+    _TOKEN = None
+
+    _HEADERS = {
+        'x-session-id': _SESSION_ID,
+        'x-client': 'browser'
+    }
+
+    _AGE_RATINGS_MAPPER = {
+        'ADULTS': 18,
+        'teens': 13
+    }
+
+    def _real_initialize(self):
+        ViuOTTNewBaseIE._TOKEN = self._download_json(
+            'https://um.viuapi.io/user/identity', None,
+            headers={'Content-type': 'application/json', **self._HEADERS},
+            query={**self._BASE_QUERY, 'iid': self._DEVICE_ID},
+            data=json.dumps({'deviceId': self._DEVICE_ID}).encode(),
+            note='Downloading token information')['token']
+
+
+class ViuOTTNewIE(ViuOTTNewBaseIE):
```

Comment:
If there is only one class, baseclass isnt needed

Reply:
Calling IEs New is a bad idea. Maybe `ViuOTTIE`?

Reply:
> If there is only one class, baseclass isnt needed

BaseClass is used to prepare possible reuseable function such as token-grabbing, header/token update, etc. 

> Calling IEs New is a bad idea. Maybe `ViuOTTIE`?

`ViuOTTIE` already defined, perhaps `ViuOTTIndonesiaIE` ? I don't sure it is restricted to Indonesia or it can accessed outside Indonesia, `ViuOTTIE` and this extractor have slightly different url

Reply:
> perhaps `ViuOTTIndonesiaIE`

Sounds good 

> > If there is only one class, baseclass isnt needed
> 
> BaseClass is used to prepare possible reuseable function such as token-grabbing, header/token update, etc.

Are you planning to add more IEs that use this?